apph-front/src/static/components/DownloadZip.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useState } from 'react';
+import { AlertColor, Box, Tooltip } from '@mui/material';
+import PhotoService from '../../services/PhotoService';
+import { AlertSnackbar } from './AlertSnackbar';
+import { IMessage, IPhoto } from '../../utils';
+import CloudDownloadIcon from '@mui/icons-material/CloudDownload';
+import { LoadingButton } from '@mui/lab';
+
+export const DownloadZip = ({
+  ids,
+  titleZip
+}: {
+  ids: number[];
+  titleZip: string;
+}): JSX.Element => {
+  const [message, setMessage] = useState('');
+  const [snackbarOpen, setSnackbarOpen] = useState<boolean>(false);
+  const [severity, setSeverity] = useState<AlertColor>();
+  const [loading, setLoading] = useState<boolean>(false);
+
+  const handleSubmit = () => {
+    setLoading(true);
+    if (ids.length != 0) {
+      downloadImage();
+    } else {
+      setSnackbarOpen(true);
+      setMessage('Aucune photo sélectionnée');
+      setSeverity('warning');
+      setLoading(false);
+    }
+  };
+  const downloadImage = () => {
+    PhotoService.downloadZip(
+      ids,
+      titleZip,
+      (photos: IPhoto) => {
+        const imageBase64 = `data:application/zip;base64,${photos.data}`;
+        const a = document.createElement('a');
+        const event = new MouseEvent('click');
+        a.href = imageBase64;
+        a.download = photos.title + photos.format;
+        a.dispatchEvent(event);
```

Comment:
c'est quoi exactement a ? 
Faudrait un nom de constante qui a du sens.

Reply:
HTML <a> Tag `https://www.w3schools.com/tags/tag_a.asp`

Reply:
```suggestion
        const link = document.createElement('a');
        const event = new MouseEvent('click');
        link.href = imageBase64;
        link.download = photos.title + photos.format;
        link.dispatchEvent(event);
```
je propose quelque chose comme ça si ça te va ou alors `downloadLink` si on veut être plus précis.